### PR TITLE
Add enum_trusted_locations.rb

### PR DESCRIPTION
**Pull request by vysecurity**
"A trusted location is typically a folder on your hard disk or a network share. Any file that you put in a trusted location can be opened without being checked by the Trust Center security feature." - https://support.office.com/en-us/article/Create-remove-or-change-a-trusted-location-for-your-files-f5151879-25ea-4998-80a5-4208b3540a62

This adds a post exploitation module to enumerate all trusted locations for different office software on a machine.
## Verification
- [ ] Start `msfconsole`
- [ ] Get a shell
- [ ] `use post/windows/gather/enum_trusted_locations`
- [ ] `set SESSION 1`
- [ ] `run`
## Output

```
msf post(enum_trusted_locations) > run

[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Found Office:
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Version found: 15.0
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Found applications.
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Found trusted locations.
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Description: Access default location: Wizard Databases
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Office15\ACCWIZ\, AllowSub: False
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Found trusted locations.
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:03] 192.168.100.53:49182 - Description: 3
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Office15\XLSTART\, AllowSub: True
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Description: 4
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Path: %APPDATA%\Microsoft\Excel\XLSTART, AllowSub: False
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Description: 5
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Path: %APPDATA%\Microsoft\Templates, AllowSub: False
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Description: 6
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Templates\, AllowSub: True
[*] [2016.06.12-11:07:04] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:05] 192.168.100.53:49182 - Description: 7
[*] [2016.06.12-11:07:05] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Office15\STARTUP\, AllowSub: True
[*] [2016.06.12-11:07:05] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:05] 192.168.100.53:49182 - Description: 12
[*] [2016.06.12-11:07:05] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Office15\Library\, AllowSub: True
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Found trusted locations.
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Description: 8
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Path: %APPDATA%\Microsoft\Templates, AllowSub: True
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Description: 9
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Templates\, AllowSub: True
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Description: 10
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - Path: %APPDATA%\Microsoft\Addins, AllowSub: False
[*] [2016.06.12-11:07:06] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:07] 192.168.100.53:49182 - Description: 11
[*] [2016.06.12-11:07:07] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Document Themes 15\, AllowSub: True
[*] [2016.06.12-11:07:07] 192.168.100.53:49182 - Found trusted locations.
[*] [2016.06.12-11:07:07] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:07] 192.168.100.53:49182 - Description: 0
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Path: %APPDATA%\Microsoft\Templates, AllowSub: False
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Description: Word 2013 default location: Application Templates
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Path: C:\Program Files\Microsoft Office\Templates\, AllowSub: True
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Description: 2
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Path: %APPDATA%\Microsoft\Word\Startup, AllowSub: False
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Description: 3
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - Path: C:\Users\Public\Templates, AllowSub: False
[*] [2016.06.12-11:07:08] 192.168.100.53:49182 - 
[*] [2016.06.12-11:07:09] 192.168.100.53:49182 - Description: 
[*] [2016.06.12-11:07:09] 192.168.100.53:49182 - Path: C:\Users\vysec\Desktop\, AllowSub: False
[+] [2016.06.12-11:07:09] 192.168.100.53:49182 - Results stored in: /root/.msf4/loot/20160612110709_default_192.168.100.53_host.trusted_loc_581583.txt
[*] Post module execution completed


```

Quickly enumerates trusted locations for file planting :)

### CODE REVIEW

**Comment by vysec** [comment]
As always, I am sure there are code improvemnets you guys have for me :)

**Comment by vysec** [comment]
Fixed some msftidy issues.

**Comment by sempervictus** [comment]
Thank you, looks neat, will test this out.
It might be useful to expand this to different targets such as Java or other common clients. With a bit of crafty mitm and local certificate store abuse, you can get some very useful credentials passed to these trusted endpoints.

**Comment by vysec** [comment]
For future I am looking at modules such as:

post/windows/gather/av/enum_comodo
post/windows/gather/av/enum_mcafee
post/windows/gather/enum_srp_policy

post/windows/gather/enum_applocker?

File: modules/post/windows/gather/enum_trusted_locations.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                  end
+              end
+            end
+        end
+      end
+      path = store_loot('host.trusted_locations', 'text/plain', session, reg_keys.join("\r\n"), 'trusted_locations.txt', 'Trusted Locations')
+      print_good("Results stored in: #{path}")
+    end
+  end
+end
```
Comment (wvu-r7) [inline modules/post/windows/gather/enum_trusted_locations.rb]: This is the end, beautiful friend / This is the end, my only friend, the end

**Comment by vysec** [comment]
haha @wvu-r7 yes it is the end.

File: modules/post/windows/gather/enum_trusted_locations.rb (hunk 1 above)
Comment (vysec) [inline modules/post/windows/gather/enum_trusted_locations.rb]: or hopefully not as I want to write more :D

Comment (wvu-r7) [inline modules/post/windows/gather/enum_trusted_locations.rb]: Maybe we can refactor the level of nesting here. :)

Comment (vysec) [inline modules/post/windows/gather/enum_trusted_locations.rb]: my Ruby sucks but any advice hugely appreciated.

**Comment by Meatballs1** [comment]
Some tidyup https://github.com/vysec/metasploit-framework/pull/1

**Comment by vysec** [comment]
Thanks Meatballs :)

**Pull request merged**